DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Repositories/SchoolRepository.cs

```diff
@@ -19,4 +20,30 @@ public class SchoolRepository(IAcademiesDbContext academiesDbContext) : ISchoolR
                     : SchoolCategory.LaMaintainedSchool))
             .SingleOrDefaultAsync();
     }
+
+    public async Task<SchoolDetails?> GetSchoolDetailsAsync(int urn)
+    {
+       return await academiesDbContext.GiasEstablishments
+            .Where(e => e.Urn == urn)
+            .Select(establishment => new SchoolDetails(establishment.EstablishmentName!,
+                stringFormattingUtilities.BuildAddressString(
+                    establishment.Street,
+                    null,
+                    establishment.Town,
+                    establishment.Postcode
+                ),
+                establishment.GorName!,
+                establishment.LaName!,
+                establishment.PhaseOfEducationName!,
+                new AgeRange(establishment.StatutoryLowAge!, establishment.StatutoryHighAge!),
+                establishment.NurseryProvisionName))
+            .SingleOrDefaultAsync();
```

Comment:
This should never be called with a urn that hasn't already been confirmed to exist so we could use `SingleAsync` and remove all the nulls and null handling code that cascades from this